docs/source/guides/grpc.rst

```diff
@@ -0,0 +1,1382 @@
+           }
+           columns {
+             int32_values: 40
+             int32_values: 190
+           }
+           columns {
+             int32_values: 140
+             int32_values: 178
+           }
+         }
+
+:bdg-primary:`API reference:` :meth:`bentoml.io.PandasDataFrame.from_proto`
+
+File-like object via ``File``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+:bdg-info:`Description:` ``File`` represents any arbitrary file type. this can be used
+to send in any file type, including images, videos, audio, etc.
+
+.. note::
+
+   Currently both :class:`bentoml.io.File` and :class:`bentoml.io.Image` are using
+   ``pb.File``
+
+It accepts the following fields:
+
+* `content`
+
+  A `bytes` field that represents the content of the file.
+
+.. TODO::
+
+   - Document ``kind`` once enum was dropped.
+   - Demonstrate python API to protobuf representation
+
+
+Complex payload via ``Multipart``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+:bdg-info:`Description:` ``Multipart`` represents a complex payload that can contain
+multiple different fields. It takes a ``fields``, which is a dictionary of input name to
+its coresponding :class:`bentoml.io.IODescriptor`
+
+.. grid:: 2
+
+    .. grid-item-card::  ``Python API``
+
+      .. code-block:: python
+
+         Multipart(
+            meta=Text(),
+            arr=NumpyNdarray(
+               dtype=np.float16,
+               shape=[2,2]
+            )
+         )
+
+    .. grid-item-card::  ``pb.Multipart``
+
+      .. code-block:: none
+
+         multipart {
+            fields {
+               key: "arr"
+               value {
+                  ndarray {
+                  dtype: DTYPE_FLOAT
+                  shape: 2
+                  shape: 2
+                  float_values: 1.0
+                  float_values: 2.0
+                  float_values: 3.0
+                  float_values: 4.0
+                  }
+               }
+            }
+            fields {
+               key: "meta"
+               value {
+                  text {
+                  value: "nlp"
+                  }
+               }
+            }
+         }
+
+:bdg-primary:`API reference:` :meth:`bentoml.io.Multipart.from_proto`
+
+Compact data format via ``serialized_bytes``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+We introduce the field ``serialized_bytes`` to both ``Request`` and ``Response`` such
+that the payload is serialized with BentoML's internal serialization format.
+
+This is useful to when we want to send a large amount of data on the wire.
+However, as mentioned above, this is an internal serialization format and thus not
+**recommended** for use by users.
+
+Mounting Servicer
+~~~~~~~~~~~~~~~~~
+
+With support for :ref:`multiplexing <guides/grpc:Demystifying the misconception of gRPC vs. REST>`
+to eliminate :wiki:`head-of-line blocking <Head-of-line_blocking>`,
+gPRC enables us to mount additional custom servicess alongside with BentoService,
+and serve them under the same port.
```

Comment:
Mounting services is independent of multiplexing and head of line blocking. Only that gRPC multiplexing makes it nicer so that one service does not block another service.

Also, there is a typo in "servicess".

Reply:
Isn't that the purpose of service multiplexing? is that multiple services can be served under the same port?

Reply:
You can technically do that in http/1 without multiplexing, like uvicorn.

Reply:
but i guess it is not support ootb